coderd/provisionerdserver/provisionerdserver_test.go

```diff
@@ -125,17 +127,9 @@ func TestHeartbeat(t *testing.T) {
 	require.True(t, ok, "first heartbeat not received")
 	_, ok = <-heartbeatChan
 	require.True(t, ok, "second heartbeat not received")
+	// Cancel the context. This should cause heartbeatDone to be closed.
 	cancel()
-	// Close the channel to ensure we don't receive any more heartbeats.
-	// The test will fail if we do.
-	defer func() {
-		if r := recover(); r != nil {
-			t.Fatalf("heartbeat received after cancel: %v", r)
-		}
-	}()
-
-	close(heartbeatChan)
-	<-time.After(testutil.IntervalMedium)
+	<-heartbeatDone
```

Comment:
what's so important about testing that heartbeats stop?

If we're just worried about leaking the heartbeat goroutine, then use `goleak` and call it a day instead of this business.

Reply:
Just that - wanted to make sure the routine wasn't leaked, and it looks like there's already a `VerifyTestMain` in this package.